### ISSUE TITLE
Question on named-functions-in-promises

### ISSUE THREAD

**Issue by sudowork** [netguru/eslint-plugin-ember#63]
So I understand the main motivation for `named-functions-in-promises` is testability. I also prefer named functions for readability in stacktraces/debugging.

However, I currently am of the opinion that using arrow functions to bind context is a bit more readable than just doing a `.bind()`. Adapting the example from the docs:

```javascript
export default Component.extend({
  actions: {
    updateUser(user) {
      user.save()
        .then(user => this._reloadUser(user))
        .then(() => this._notifyAboutSuccess())
        .catch(err => this._notifyAboutFailure(err));
        // .then(this._reloadUser.bind(this))
        // .then(this._notifyAboutSuccess.bind(this))
        // .catch(this._notifyAboutFailure.bind(this));
    },
  },

  // ... Imagine there's a lot of other functions here ...

  _reloadUser(user) {
    return user.reload();
  },
  _notifyAboutSuccess() { /* ... */ },
  _notifyAboutFailure(err) { /* ... */ },
});
```

From my perspective, this still keeps the testability and debuggability (though it will add an extra call to an anonymous function). It has the added benefit of being able to read through the promise chain and know what parameters each function takes without having to jump up and down the file. I'm happy to be persuaded otherwise.

Currently, the rule disallows using the anonymous functions, including arrow function expressions. My proposal would be to also allow arrow function expressions where the body is a `CallExpression`. To be more strict, can ensure that the method being called belongs to the parent `this`. Thoughts?

**Comment by bardzusny** [comment]
Thanks for your PR! 🙂 

This one comes back once in a while 😄 . I believe it's a good direction to go in, while not being 100% convinced that allowing only methods defined on `this` is the right idea. Most importantly, current rule perfectly allows i.e.:

```javascript
this._loadUser()
  .then(somethingEntirelyFromTheOutside)
  .then(...);
```

... and breaking away with this seems both unnecessary and backwards-incompatible.

TL;DR I think that simply allowing one-line arrow functions as promise callbacks is the best option, likely as additional rule configuration option. WDYT?

**Comment by bardzusny** [comment]
I'll also leave here an example of case that I met with several times IRL, and would be solved by this proposed modification to `named-functions-in-promises` rule. Namely: passing additional parameters to promise callbacks.

```javascript
export default Component.extend({
  getUser(id) {
    return get(this, 'store')
      .findRecord('user', id)
      .then(this._formatUser.bind(this, 'admin'));
  },

  _formatUser(userType, user, someExtraOptions) {
    ...
  },
});
```

See how:
* weirdish `.bind` call is necessary to pass `userType` to `_formatUser` method
* `someOptionalParam` is hard to pass, at all - it would require changing `_formatUser` method parameters order.

How it could look like instead:
```
// before:
.then(this._formatUser.bind(this, 'admin'));
// after:
.then((user) => this._formatUser('admin', user, { someExtraOptions }));
```

**Comment by sudowork** [comment]
Yes, I've also definitely run into the `.bind(this, partially, applied)` and find the constraint on parameter ordering to be restrictive.

I'm happy to remove the check that the method must be called on `this`. I was trying to read between the lines in the docs/example and I thought that may have been what was desired.

**Comment by michalsnik** [comment]
Okay, I left some comments in your PR. I think we're almost there @sudowork  :)
